### DISCUSSION TITLE
Support react-native-web via swc transform

### DISCUSSION THREAD

**Discussion by merrywhether** [vercel/next.js#36566] [Ideas]
### Describe the feature you'd like to request

[react-native-web](https://necolas.github.io/react-native-web/) is becoming increasingly popular to pair with Next, with package like [Solito](https://solito.dev) making it even easier. It's also included in the [official examples list](https://github.com/vercel/next.js/tree/canary/examples/with-react-native-web). CRA includes built-in support for RNW as well.

But using RNW forces you to opt out of using `swc` because of the need for [babel-plugin-react-native-web](https://www.npmjs.com/package/babel-plugin-react-native-web) and thus a `babel.config.js`. Similar to the work done for styled-components and relay, an swc transform for this relatively lightweight plugin would be great for those adding RN/RNW to their workflows.

### Describe the solution you'd like

Built-in support for swc tranforms of react-native to react-native-web components.

### Describe alternatives you've considered

Learning Rust and doing this myself does sound fun, but I'm not quite there yet. 😄 Otherwise, continuing to use the babel transform works, just more slowly.

**Comment by nissy-dev** [comment]
I created swc plugin for react-native-web 🚀 
https://github.com/nissy-dev/swc-plugin-react-native-web

Recently, Next.js supports custom swc plugins (see: https://github.com/vercel/next.js/pull/36790), so you can try this plugin with Next.js canary version. I made [example project](https://github.com/nissy-dev/swc-plugin-react-native-web/tree/main/example) for Next.js and confirmed it works well.
